src/olympia/addons/admin.py

```diff
@@ -258,44 +259,25 @@ class Media(AMOModelAdmin.Media):
     )
     actions = ['git_extract_action']
 
-    def get_queryset(self, request):
-        # We want to _unlisted_versions_exists/_listed_versions_exists to avoid
-        # repeating that query for each add-on in the list. A cleaner way to do this
-        # would be to use annotate like this:
-        # sub_qs = Version.unfiltered.filter(addon=OuterRef('pk')).values_list('id')
-        # (...).annotate(
-        #     _unlisted_versions_exists=Exists(
-        #         sub_qs.filter(channel=amo.CHANNEL_UNLISTED)
-        #     ),
-        #     _listed_versions_exists=Exists(
-        #         sub_qs.filter(channel=amo.CHANNEL_LISTED)
-        #     ),
-        # )
-        # But while this works, the subquery is a lot less optimized (it does a full
-        # query instead of the SELECT 1 ... LIMIT 1) and to make things worse django
-        # admin doesn't know it's only for displayed data (it doesn't realize we aren't
-        # filtering on it, and even if it did can't remove the annotations from the
-        # queryset anyway) so it uses it for the count() queries as well, making them a
-        # lot slower.
```

Comment:
That comment was obsolete: not only does Django optimizes the subquery nowadays, it was also executing the `extra()` because the way `.count()` works has changed since that comment was written.

Which means there was no reason to do it without using annotations anymore. Moving it to the newly created `get_queryset_annotations()` allows us to avoid it for the `COUNT(*)` query.